### PR TITLE
Dom utils refactor

### LINKED ISSUE #2259: Dom Utils refactor
Unused methods, etc

### PR DESCRIPTION
**Pull request by aicioara**
Closes #2259

Changes:
- Things that can inferred (like return type of methods) are omitted now
- Style cleanup
- `getBBox` -> `elementBBox`
- `getParsedStyleValue` -> REMOVED (privatized)
- `isSelectionRemovedFromSVG` -> REMOVED
- `getElementWidth` -> `elementWidth`
- `getElementHeight` -> `elementHeight`
- `getSVGPixelWidth` -> REMOVED
- `getBoundingSVG` -> `boundingSVG`
- `getUniqueClipPathId` -> `generateUniqueClipPathId`

@bluong EDIT: second `getElementWidth` -> `getElementHeight`

**QE Notes** Regression pass

### CODE REVIEW

File: src/utils/domUtils.ts [hunk 1]
```diff
@@ -9,99 +9,58 @@ export module Utils {
      * @param {d3.Selection} element
      * @returns {SVGRed} The bounding box.
      */
-    export function getBBox(element: d3.Selection<any>): SVGRect {
+    export function getBBox(element: d3.Selection<any>) {
```
Comment (bluong) [inline src/utils/domUtils.ts]: Are we keeping `getBBox()`?  It reflects the endpoint name, but it's inconsistent now.  Not sure...

Maybe `elementBBox()`?

File: src/utils/domUtils.ts [hunk 2]
```diff
@@ -9,99 +9,58 @@ export module Utils {
...
+        bbox = { x: 0, y: 0, width: 0, height: 0 };
       }
       return bbox;
     }
 
-    export var POLYFILL_TIMEOUT_MSEC = 1000 / 60; // 60 fps
-    export function requestAnimationFramePolyfill(fn: () => any): void {
+    export var POLYFILL_TIMEOUT_MILLISECONDS = 1000 / 60; // 60 fps
+
+    export function requestAnimationFramePolyfill(fn: () => any) {
```
Comment (bluong) [inline src/utils/domUtils.ts]: Now that we decided to expose these things.

Do you think `requestAnimationFramePolyfill` is a good name?

Comment (bluong) [inline src/utils/domUtils.ts]: Also, `() => any` should be renamed to `Function`

File: src/utils/domUtils.ts [hunk 3]
```diff
@@ -9,99 +9,58 @@ export module Utils {
...
     }
 
-    export function translate(s: d3.Selection<any>): d3.Transform;
-    export function translate(s: d3.Selection<any>, x: number, y: number): d3.Selection<any>
-    export function translate(s: d3.Selection<any>, x?: number, y?: number): any {
-      var xform = d3.transform(s.attr("transform"));
+    export function translate(selection: d3.Selection<any>): d3.Transform;
+    export function translate(selection: d3.Selection<any>, x: number, y: number): d3.Selection<any>;
+    export function translate(selection: d3.Selection<any>, x?: number, y?: number): any {
+      var xform = d3.transform(selection.attr("transform"));
```
Comment (bluong) [inline src/utils/domUtils.ts]: Out of scope, but I was really thrown off by the variable `xform`

Comment (aicioara) [inline src/utils/domUtils.ts]: lol

File: src/utils/domUtils.ts (hunk 2 above)
Comment (aicioara) [inline src/utils/domUtils.ts]: `error TS2345: Argument of type 'Function' is not assignable to parameter of type 'FrameRequestCallback'.
`

Comment (aicioara) [inline src/utils/domUtils.ts]: In my understanding, it is consistent with `window.requestAnimationFrame()`

Comment (bluong) [inline src/utils/domUtils.ts]: does `() => void` work?

**Comment by bluong** [comment]
I think this is NOQE?

**Comment by aicioara** [comment]
I'd give it a quick regression pass. I checked quicktests myself

**Pull request merged**